packages/theme/src/cli/commands/theme/check.ts

```diff
@@ -97,8 +125,81 @@ Excludes checks matching any category when specified more than once`,
 
   async run(): Promise<void> {
     const {flags} = await this.parse(Check)
-    await execCLI2(['theme', 'check', flags.path, ...this.passThroughFlags(flags, {allowedFlags: Check.cli2Flags})], {
-      directory: flags.path,
+
+    // Its not clear to typescript that path will always be defined
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+    const path = flags.path!
+
+    if (flags['dev-preview']) {
+      if (flags.init) {
+        await initConfig(path)
+
+        // --init should not trigger full theme check operation
+        return
+      }
+
+      if (flags.version) {
+        outputInfo(themeCheckPackage.version)
+
+        // --version should not trigger full theme check operation
+        return
+      }
+
+      if (flags.print) {
+        await outputActiveConfig(flags.config, path)
+
+        // --print should not trigger full theme check operation
+        return
+      }
+
+      if (flags.list) {
+        await outputActiveChecks(flags.config, path)
+
+        // --list should not trigger full theme check operation
+        return
+      }
+
+      const {offenses, theme} = await themeCheckRun(path, flags.config)
+
+      const offensesByFile = sortOffenses(offenses)
+
+      if (flags.output === 'text') {
+        renderOffensesText(offensesByFile, path)
+
+        renderInfo({
```

Comment:
Personally, I believe it would be nice if we could invoke `renderSuccess` when there are no offenses.

![image](https://github.com/Shopify/cli/assets/1079279/c1e709e3-6fbc-47ee-82e2-3a1573c1408e)

From a user's perspective, I usually appreciate the quick, rewarding green feedback when everything is fixed :)